subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -57,3 +57,117 @@ To fix this problem, you must remove the property and use the `TransformOutputs`
+
+For example, you wrote:
+
+```groovy
+class MyTask extends DefaultTask {
+    private Property<Integer> x
+
+    @Input
+    Property<Integer> getX() { this.x }
+
+    void setX(Property<Integer> x) { this.x = x }
+}
+```
+
+However, mutable types like `Property` are designed to track dependencies and origin of values they are supplied with.
+As a consequence, it is an error to make them overwritable, because Gradle will then be unable to tell where a property value comes from.
+
+To fix this problem, you should either make your property final and remove the setter:
+
+```groovy
+class MyTask extends DefaultTask {
+    @Input
+    final Property<Integer> getX() = ...
+```
+
+or rely on Gradle's built-in capacity to inject final properties:
+
+```groovy
+abstract class MyTask {
+    abstract Property<Integer> getX()
+}
+```
+
+then the value of the properties need to be configured via the mutation methods:
+
+```groovy
+myTask {
+    x.set(123)
+}
+```
+
+[[redundant_getters]]
+== Redundant getters
+
+This error indicates that a `boolean` property has both a `get` and an `is` getter method.
+This is a problem because both getters can be annotated differently and Gradle cannot know which ones to use.
+
+The solution to this problem is to get rid of one of the getters.
+
+[[private_getter_must_not_be_annotated]]
+== Annotations on private getters
+
+This error indicates that you have annotated a _private_ getter with an input or output annotation.
+Gradle doesn't use private getters as inputs for up-to-date checking, which means that your effectively are basically ignored.
+It is important to fix because you might think that you have declared an input when it's not the case.
+
+To fix this, either make the getter public, or annotate an existing getter instead, or create a new annotated getter.
+
+[[ignored_property_must_not_be_annotated]]
+== Annotations on ignored properties
+
+This error indicates that you have a property which is annotated with an annotation which tells Gradle to ignore it (for example link:{javadocPath}/org/gradle/api/model/ReplacedBy.html[`@ReplacedBy`]) but is also annotated with an input annotation (for example link:{javadocPath}/org/gradle/api/tasks/InputFile.html[`@InputFile`]).
+
+This is an error because Gradle cannot determine if the property should actually be used for up-to-date checking, that is to say if it's actually an input or not.
+
+To fix this, you must either:
+
+- remove the input annotations from the property or
+- remove the ignoring annotation from the property
+
+[[conflicting_annotations]]
+== Conflicting annotations
+
+This error indicates that a property is annotated with conflicting annotations, that is to say annotations which have different, irreconciliable semantics.
+
+For example, a property cannot be annotated both with `@InputFile` and `@InputArtifact` at the same time.
```

Comment:
I think `@InputFile` and `@OutputFile` is better, since most folks don't know about `@InputArtifact`.